www/js/draw-model.js

```diff
@@ -479,6 +492,47 @@ define([
 		updateNodeConnections: function(from, to){
 			this.detachConnections(to);
 			this.setConnections(from, to);
+		},
+		getStatus: function(node){
+			var nodeStatus = node.status;
+			var nodeAttributesNumberExpected;
+			var variableType = this._model.getVariableType(node.ID);
+			var modelType = this._model.getType(node.ID);
+			if(variableType && modelType){
+				if(modelType === "quantity"){
+					if(variableType === "dynamic" || variableType === "parameter"){
+						nodeAttributesNumberExpected = 5;
+					}else{
+						nodeAttributesNumberExpected = 4;
+					}
+				}
+				if(modelType === "equation"){
+					nodeAttributesNumberExpected = 2;
+				}
+				var nodeAttributesNumber = Object.keys(nodeStatus).length;
+				var attributeCounter = 0;
+				for(var attribute in nodeStatus){
+					var _status = nodeStatus[attribute].status; 
+					if(_status !== undefined && _status === "correct"){ 
+						attributeCounter++;
+					}else if(_status !== undefined && _status === "incorrect"){
+						attributeCounter = -nodeAttributesNumber+1;
+					}
+				}
+				var nodeClass = "";
+				// TO DO :Consider attempts
+				if( attributeCounter === 0){
+					nodeClass = " fa-minus";	
+				}else if( attributeCounter !== 0 && attributeCounter === nodeAttributesNumber && nodeAttributesNumber === nodeAttributesNumberExpected /*Add attempt count here*/){
+					nodeClass = " fa-star";
+				}else if( attributeCounter < 0){
+					nodeClass = " fa-times";
+				}else{
+					nodeClass = " fa-check";
+				}
+				return nodeClass;
+
```

Comment:
I dont think this code should be here. Draw-model is not supposed to take decision that is the conditions are decided by pedagogical-module and it should tell everyone else constructor or draw-model that show this class. So there can be a map which will tell the class and it will ask draw-model to go do it, same goes for constructor.